packages/sdk/src/dataProtector/sharing/setProtectedDataToRenting.ts

```diff
@@ -1,91 +1,116 @@
-import { WorkflowError } from '../../utils/errors.js';
-import { throwIfMissing } from '../../utils/validators.js';
+import { GraphQLClient } from 'graphql-request';
+import { DEFAULT_SHARING_CONTRACT_ADDRESS } from '../../config/config.js';
+import { ErrorWithData, WorkflowError } from '../../utils/errors.js';
+import {
+  addressOrEnsOrAnySchema,
+  positiveNumberSchema,
+  positiveStrictIntegerStringSchema,
+  throwIfMissing,
+} from '../../utils/validators.js';
 import {
   IExecConsumer,
   SetProtectedDataToRentingParams,
   SuccessWithTransactionHash,
   SubgraphConsumer,
+  Address,
 } from '../types/index.js';
 import { getSharingContract } from './smartContract/getSharingContract.js';
-import {
-  collectionExists,
-  isCollectionOwner,
-  isProtectedDataInCollection,
-} from './utils.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
 
 export const setProtectedDataToRenting = async ({
   iexec = throwIfMissing(),
   graphQLClient = throwIfMissing(),
-  collectionTokenId = throwIfMissing(),
   protectedDataAddress = throwIfMissing(),
   priceInNRLC = throwIfMissing(),
   durationInSeconds = throwIfMissing(),
 }: IExecConsumer &
   SubgraphConsumer &
   SetProtectedDataToRentingParams): Promise<SuccessWithTransactionHash> => {
-  //TODO:Input validation
+  const vProtectedDataAddress = addressOrEnsOrAnySchema()
+    .required()
+    .label('protectedDataAddress')
+    .validateSync(protectedDataAddress);
+  const vPriceInNRLC = positiveNumberSchema()
+    .required()
+    .label('priceInNRLC')
+    .validateSync(priceInNRLC);
+  const vDurationInSeconds = positiveStrictIntegerStringSchema()
+    .required()
+    .label('durationInSeconds')
+    .validateSync(durationInSeconds);
```

Comment:
The more I look at it, the more I think we should rename it to `durationInDays`. It's gonna be annoying for builders to always convert to seconds. Or at least in hours?

We can keep it in seconds for the smart-contract.

Reply:
yeah we can make a refacto in an other PR for that. But in the SC it should stay in second